cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,39 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
+  let user;
   before(() => {
-    
+    cy.visit('/');
+    cy.task('generateUser').then(generatedUser => {
+      user = generatedUser;
+      user.birthDate = '05 August,1978';
+    });
   });
 
-  it('', () => {
-    
+  it('should allow to register', () => {
+    cy.viewport(550, 750);
+    cy.findByPlaceholder('First Name').type(user.firstName);
+    cy.findByPlaceholder('Last Name').type(user.lastName);
+    cy.get('#userEmail').type(user.email);
+    cy.findByType('radio').check(user.gender, { force: true });
+    cy.findByPlaceholder('Mobile Number').type(user.mobileNumber);
+    cy.get('#dateOfBirthInput').type('{selectAll} ' + user.birthDate + '{esc}');
+    cy.get('.subjects-auto-complete__value-container').type('Com{enter}');
```

Comment:
Avoid using hardcoded values like 'Com{enter}' while typing. Instead, use a variable to store the subject and type it.